From 62ce211d5548b67afae0741abbd304706dc09f8b Mon Sep 17 00:00:00 2001
From: Jiebing Li <jiebing.li@intel.com>
Date: Mon, 19 Jan 2015 17:59:31 +0800
Subject: [PATCH 1/1] usb: android gadget: add 900mA Max Power support for
 SuperSpeed

This patch adds 900mA maximum power consuption support for
SuperSpeed mode. Please notice, as it is 8mA per unit in
descriptor, so the real max value is 896mA instead of 900mA.

Change-Id: Ic2667d4bdd6f952657e7a24a8286c99a4eef10f0
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-16223
Signed-off-by: Jiebing Li <jiebing.li@intel.com>
Signed-off-by: Wu, Hao <hao.wu@intel.com>
---
 drivers/usb/gadget/android.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index a516896..bf66f9c 100644
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -1558,6 +1558,15 @@ android_setup(struct usb_gadget *gadget, const struct usb_ctrlrequest *c)
 	spin_lock_irqsave(&cdev->lock, flags);
 	if (!dev->connected) {
 		dev->connected = 1;
+
+		/* set MaxPower as 896mA (8mA per unit in configuration
+		 * descriptor bMaxPower, and < 900mA per USB3 spec) for
+		 * SuperSpeed mode */
+		if (gadget->speed == USB_SPEED_SUPER)
+			android_config_driver.MaxPower = 896;
+		else
+			android_config_driver.MaxPower = 500;
+
 		schedule_work(&dev->work);
 	} else if (c->bRequest == USB_REQ_SET_CONFIGURATION &&
 						cdev->config) {
-- 
1.9.1

